### PR TITLE
add json, text, stderr args

### PR DESCRIPTION
**Pull request by chylli-binary**
https://redmine.deriv.cloud/issues/34250#logs/refactor-log-any-adapter-deriv

### CODE REVIEW

**Review by shantanu-deriv — APPROVED** [review]

File: lib/Log/Any/Adapter/DERIV.pm [hunk 1]
```diff
@@ -56,6 +69,28 @@ their own version or make feature/bugfix suggestions if they seem generally usef
...
+If it is true, then print logs to STDERR
+
+If the value is json or text, then print logs with that format
+
+If the value is just a true value other than `json` or `text`, then if it is running in a container, 
+then the logs is `json` format. Else if STDERR is a tty will be `colored text` format. Else if will be a non-color text format.
+
+=back
+
+If no any paramter, then default `stderr => 1`;
```
Comment (shantanu-deriv) [inline lib/Log/Any/Adapter/DERIV.pm:92]:
```suggestion
If no any parameter, then default `stderr => 1`;
```

Comment (shantanu-deriv) [reply lib/Log/Any/Adapter/DERIV.pm:92]: This will resolve the failing circle ci pod test

File: lib/Log/Any/Adapter/DERIV.pm [hunk 2]
```diff
@@ -144,18 +179,21 @@ $SIG{__DIE__} = sub {
...
-    if($json_log_file) {
-        $self->{fh} = path($json_log_file)->opena_utf8 or die 'unable to open log file - ' . $!;
-        $self->{fh}->autoflush(1);
+    # if there is json_log_file, then print josn to that file
+    if($self->{json_log_file}) {
+        $self->{json_fh} = path($self->{json_log_file})->opena_utf8 or die 'unable to open log file - ' . $!;
+        $self->{json_fh}->autoflush(1);
     }
+    # if there is stderr, then print log to stderr also
+    # if stderr is josn or text, then use that format
```
Comment (sasikala-binary) [inline lib/Log/Any/Adapter/DERIV.pm:190]:
```suggestion
    # if stderr is json or text, then use that format
```

File: lib/Log/Any/Adapter/DERIV.pm [hunk 3]
```diff
@@ -56,6 +69,28 @@ their own version or make feature/bugfix suggestions if they seem generally usef
...
+Specify a file name that the json format log file will be printed into.
+If not given, then a default file 'program_name.json.log' will be used.
+
+=item * STDERR
+
+If it is true, then print logs to STDERR
+
+If the value is json or text, then print logs with that format
+
+If the value is just a true value other than `json` or `text`, then if it is running in a container,
```
Comment (chylli-binary) [inline lib/Log/Any/Adapter/DERIV.pm:87]:
```suggestion
If the value is just a true value other than `json` or `text`, then if it is running in a container,
```

File: lib/Log/Any/Adapter/DERIV.pm [hunk 4]
```diff
@@ -144,18 +181,21 @@ $SIG{__DIE__} = sub {
...
-    # is at least a starting point.
-    $self->{in_container} = -r '/.dockerenv';
-    my $json_log_file = $self->{json_log_file};
-    $json_log_file = $0 . '.json.log' if(!$json_log_file && !$self->{in_container});
-    if($json_log_file) {
-        $self->{fh} = path($json_log_file)->opena_utf8 or die 'unable to open log file - ' . $!;
-        $self->{fh}->autoflush(1);
+    # if there is json_log_file, then print json to that file
+    if($self->{json_log_file}) {
+        $self->{json_fh} = path($self->{json_log_file})->opena_utf8 or die 'unable to open log file - ' . $!;
```
Comment (sasikala-binary) [inline lib/Log/Any/Adapter/DERIV.pm:188]: Before open file could we check for existence,  if the file is not there then touch and give permission here?

Comment (chylli-binary) [reply lib/Log/Any/Adapter/DERIV.pm:188]: We needn't. It is opened with `append` mode. So if there is one, then open it , if not, it will be created.

**Review by sasikala-binary — APPROVED** [review]

**Pull request merged**